### DISCUSSION TITLE
Images are rendering with tiles

### DISCUSSION THREAD

**Discussion by happysurf** [upscayl/upscayl#600] [Ideas]
After looking an upscaled image with your program, I noticed with a zoom of about 600%, that the rendering had been perform with the _Tile parameter_, not set to 0.
I had noticed this problem in some online services images result in the past.

This causes the completion of the image through a series of rendering tiles which however present some very ugly interruptions which you can also notice on the print if you look carefully.

This setting decreases the load of the GPU but the result does not justify this choice in my opinion.
To understand all this, simply go to the model developer's website (Xintao) and try to upscale an image, by setting the Anime model (in my case) and leaving the _Tile parameter_ to 0.

The result will be uniform because the rendering is done only once.
The Tile setting affected also the other Xintao models, still I don't test the other Upscayl models, like ultrasharp or ultramix balanced.

I also attach the explanatory images, the first is with Upscayl (digital art), the second is created on Xintao (anime V6) web site.

Xintao page: https://replicate.com/xinntao/realesrgan?prediction=jxxbf3rb4hqk76wv5uo2ebwlia

![Upscayl](https://github.com/upscayl/upscayl/assets/13637587/f20e3dd8-9bf9-448b-961e-b53160dd9b2b)

![Xintao](https://github.com/upscayl/upscayl/assets/13637587/459fef7f-ceb0-4014-913e-53b7f9f9d78d)

**Comment by NayamAmarshe** [comment]
This is a common problem in super resolution and there's nothing we (Upscayl) can do about it for now.
The tiling is a required setting to make sure we don't go out of bounds with the GPU memory, which is extremely easy to do with a custom tile size.

When the tile size is 0 (default), it automatically sets the tile size according to the available GPU memory.

**Comment by happysurf** [comment]
This is a very bad news, so the only condition for having a single rendering is to have a very powerful graphics card, like the Nvidia T4 GPU  like in Replicate web site I think...

Some hope in the future Upscayl Cloud?

**Reply by NayamAmarshe** [reply]
Even with a GPU with huge memory, PyTorch tensors run out of memory. So I don't think it's something that can be fixed easily. Will keep looking for solutions though.
